models/dodo/ethereum/dodo_aggregator_ethereum_trades.sql

```diff
@@ -20,174 +21,175 @@ WITH dexs AS
             toToken AS token_sold_address,
             contract_address AS project_contract_address,
             evt_tx_hash AS tx_hash,
-            CAST(ARRAY() as array<bigint>) AS trace_address,
+            CAST(NULL as array<bigint>) AS trace_address,
             evt_index
         FROM
             {{ source('dodo_ethereum', 'DODOV1Proxy04_evt_OrderHistory')}}
         {% if is_incremental() %}
-        WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
         {% endif %}
 
         UNION ALL
 
         -- dodov2 proxy02
         SELECT
             evt_block_time AS block_time,
-            'DODO' AS project,
+            'DODO X' AS project,
             '0' AS version,
             sender AS taker,
-            '' AS maker,
+            CAST(NULL AS VARBINARY) AS maker,
             fromAmount AS token_bought_amount_raw,
             returnAmount AS token_sold_amount_raw,
             cast(NULL as double) AS amount_usd,
             fromToken AS token_bought_address,
             toToken AS token_sold_address,
             contract_address AS project_contract_address,
             evt_tx_hash AS tx_hash,
-            CAST(ARRAY() as array<bigint>) AS trace_address,
+            CAST(NULL as array<bigint>) AS trace_address,
             evt_index
         FROM
             {{ source('dodo_ethereum','DODOV2Proxy02_evt_OrderHistory')}}
         {% if is_incremental() %}
-        WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
         {% endif %}
 
         UNION ALL
 
         -- DODORouteProxy
         SELECT
             evt_block_time AS block_time,
-            'DODO' AS project,
+            'DODO X' AS project,
             '0' AS version,
             sender AS taker,
-            '' AS maker,
+            CAST(NULL AS VARBINARY) AS maker,
             fromAmount AS token_bought_amount_raw,
             returnAmount AS token_sold_amount_raw,
             cast(NULL as double) AS amount_usd,
             fromToken AS token_bought_address,
             toToken AS token_sold_address,
             contract_address AS project_contract_address,
             evt_tx_hash AS tx_hash,
-            CAST(ARRAY() as array<bigint>) AS trace_address,
+            CAST(NULL as array<bigint>) AS trace_address,
             evt_index
         FROM
             {{ source('dodo_ethereum','DODORouteProxy_evt_OrderHistory')}}
         {% if is_incremental() %}
-        WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
         {% endif %}
 
         UNION ALL
         
         -- DODOFeeRouteProxy
         SELECT
             evt_block_time AS block_time,
-            'DODO' AS project,
+            'DODO X' AS project,
```

Comment:
@owen05 I would suggest that you add a `proxy` field here, to distinguish records.
The `proxy` field may help you to identify where the dupes come from.

There may be overlapping records, like proxy calls each other.